@nrwl/next produces faulty build on production environment

## Current Behavior
Build output of @nrwl/next package beyond workspace version 12.4.0 comes with missing dependencies when installed and ran on `NODE_ENV=production`

## Expected Behavior
The build should run without issues using the start script generated by the builder after installation.
This **is** a regression, the build works without issues if i manually patch @nrwl/next to version 12.4.0 in the runner container.

## Steps to Reproduce
* Create a next application in a preexisting Nrwl workspace at version 12.4.0
* Migrate to 12.8.0+ (Tested this on 12.8.0, 12.9.0-rc0, 12.9.0)
* Generate a production build of the application with `nx build --prod` command.
* Using a Dockerfile, copy the contents of the dist folder into a container, set NODE_ENV=production environment variable and use `yarn` command
* Define `yarn start -p 3000` command as CMD in the same Dockerfile
* The container will start and complain about `typescript` being a missing dependency while it obviously should be a dev dependency rather than a final build dependency

### Failure Logs

```
yarn run v1.22.5
warning package.json: No license field

$ next start -p '${PORT}'

ready - started server on 0.0.0.0:3000, url: http://localhost:3000

Error: Cannot find module 'typescript'

Require stack:
- /node_modules/@nrwl/workspace/src/utilities/typescript/compilation.js
- /node_modules/@nrwl/workspace/src/utilities/typescript.js
- /node_modules/@nrwl/workspace/src/core/target-project-locator.js
- /node_modules/@nrwl/workspace/src/core/project-graph/build-dependencies/explicit-project-dependencies.js
- /node_modules/@nrwl/workspace/src/core/project-graph/build-dependencies/index.js
- /node_modules/@nrwl/workspace/src/core/project-graph/project-graph.js
- /node_modules/@nrwl/workspace/src/core/file-utils.js
- /node_modules/@nrwl/next/plugins/with-nx.js
- /next.config.js
- /node_modules/next/dist/server/config.js
- /node_modules/next/dist/server/next.js
- /node_modules/next/dist/server/lib/start-server.js
- /node_modules/next/dist/cli/next-start.js
- /node_modules/next/dist/bin/next
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:889:15)
    at Function.mod._resolveFilename (/node_modules/next/dist/build/webpack/require-hook.js:96:28)
    at Function.Module._load (internal/modules/cjs/loader.js:745:27)
    at Module.require (internal/modules/cjs/loader.js:961:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at Object.<anonymous> (/node_modules/@nrwl/workspace/src/utilities/typescript/compilation.js:7:12)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12) {

  code: 'MODULE_NOT_FOUND',

  requireStack: [
    '/node_modules/@nrwl/workspace/src/utilities/typescript/compilation.js',
    '/node_modules/@nrwl/workspace/src/utilities/typescript.js',
    '/node_modules/@nrwl/workspace/src/core/target-project-locator.js',
    '/node_modules/@nrwl/workspace/src/core/project-graph/build-dependencies/explicit-project-dependencies.js',
    '/node_modules/@nrwl/workspace/src/core/project-graph/build-dependencies/index.js',
    '/node_modules/@nrwl/workspace/src/core/project-graph/project-graph.js',
    '/node_modules/@nrwl/workspace/src/core/file-utils.js',
    '/node_modules/@nrwl/next/plugins/with-nx.js',
    '/next.config.js',
    '/node_modules/next/dist/server/config.js',
    '/node_modules/next/dist/server/next.js',
    '/node_modules/next/dist/server/lib/start-server.js',
    '/node_modules/next/dist/cli/next-start.js',
    '/node_modules/next/dist/bin/next'
  ]
}

error Command failed with exit code 1.
```

### Environment

```
  Node : 16.8.0
  OS   : win32 x64
  yarn : 1.22.5

  nx : 12.9.0
  @nrwl/angular : Not Found
  @nrwl/cli : 12.9.0
  @nrwl/cypress : 12.9.0
  @nrwl/devkit : 12.9.0
  @nrwl/eslint-plugin-nx : 12.9.0
  @nrwl/express : Not Found
  @nrwl/jest : 12.9.0
  @nrwl/linter : 12.9.0
  @nrwl/nest : Not Found
  @nrwl/next : 12.9.0
  @nrwl/node : Not Found
  @nrwl/nx-cloud : Not Found
  @nrwl/react : 12.9.0
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.9.0
  @nrwl/web : 12.9.0
  @nrwl/workspace : 12.9.0
  @nrwl/storybook : 12.9.0
  @nrwl/gatsby : Not Found
  typescript : 4.3.5
```

### References

This issue is the same as #6886, I included a bit more stuff, as I know it used to work fine and still does work on version workspace and next 12.4.0

--- a/packages/next/src/executors/build/lib/create-package-json.ts
+++ b/packages/next/src/executors/build/lib/create-package-json.ts
@@ -25,8 +25,14 @@ export async function createPackageJson(
   const nrwlWorkspaceNode = depGraph.nodes['npm:@nrwl/workspace'];
 
   if (nrwlWorkspaceNode) {
-    packageJson.devDependencies['@nrwl/workspace'] =
+    packageJson.dependencies['@nrwl/workspace'] =
       nrwlWorkspaceNode.data.version;
   }
+
+  const typescriptNode = depGraph.nodes['npm:typescript'];
+  if (typescriptNode) {
+    packageJson.dependencies['typescript'] = typescriptNode.data.version;
+  }
+
   writeJsonFile(`${options.outputPath}/package.json`, packageJson);
 }

